packages/vow/test/unhandled-rejections.test.js

```diff
@@ -0,0 +1,274 @@
+          },
+          { message: DEFAULT_REASON.message },
+        ),
+    );
+  });
+});
+
+/**
+ * A vow which is rejected, and then `shorten` is called after the vow's status
+ * has recorded the rejection. This should trigger the unhandled rejection
+ * tracker, but not result in an unhandled rejection.
+ */
+test.serial('vow rejected, then shorten', async t => {
+  annihilate();
+
+  await startLife(async baggage => {
+    const zone = makeDurableZone(baggage, 'durableRoot');
+    const vowTools = prepareVowTools(zone);
+
+    const vow = makeRejectedVow(vowTools);
+    shorten(vow).then(
+      value => t.fail(value),
+      reason =>
+        t.throws(
+          () => {
+            throw reason;
+          },
+          { message: DEFAULT_REASON.message },
+        ),
+    );
+  });
+});
+
+/**
+ * a vow which is shortened, then goes through an upgrade before being rejected,
+ * then shortened again. This should similarly trigger the unhandled rejection
+ * tracker, but not result in an unhandled rejection.
+ */
+test.serial(
+  'vow shortened, upgraded, rejected, then shortened again',
+  async t => {
+    annihilate();
+
+    await startLife(async baggage => {
+      const zone = makeDurableZone(baggage, 'durableRoot');
+      const vowTools = prepareVowTools(zone);
+
+      const testVowKit = zone.makeOnce('testVowKit', () =>
+        vowTools.makeVowKit(),
+      );
+
+      shorten(testVowKit.vow).then(
+        value => t.fail(value),
+        reason => t.fail(reason),
+      );
+    });
+
+    await startLife(async baggage => {
+      const zone = makeDurableZone(baggage, 'durableRoot');
+      prepareVowTools(zone);
+
+      const testVowKit = zone.makeOnce('testVowKit', () =>
+        t.fail('need testVowKit in baggage'),
+      );
+
+      testVowKit.resolver.reject(DEFAULT_REASON);
+
+      shorten(testVowKit.vow).then(
+        value => t.fail(value),
+        reason =>
+          t.throws(
+            () => {
+              throw reason;
+            },
+            { message: DEFAULT_REASON.message },
+          ),
+      );
+    });
+  },
+);
+
+/**
+ * A vow which is rejected without being shortened, then dropped. This should
+ * trigger the unhandled rejection promise reporting after gc.
+ */
+test.serial(expectUnhandled, 1, 'vow rejected, then dropped', async t => {
+  annihilate();
+
+  await startLife(async baggage => {
+    const zone = makeDurableZone(baggage, 'durableRoot');
+    const vowTools = prepareVowTools(zone);
+
+    makeRejectedVow(vowTools);
+    await gcAndFinalize();
+  });
+
+  await gcAndFinalize();
+  t.pass();
+});
+
+/**
+ * A vow which is rejected without being shortened, but stored in baggage. Then
+ * after an upgrade the vow is handled. This should trigger an unhandled
+ * rejection with our custom "not handled before upgrade" error, followed by the
+ * "handling prior incarnation" warning.
+ *
+ * The rejection reason *is* storable.
+ */
+test.serial(
+  expectUnhandled,
+  2,
```

Comment:
I suppose that the 2 unhandled rejections are an artifact of the test infra, and in real life we'd only expect 1 (the after upgrade one)? The promise created by the unhandled rejection tracker in the first incarnation will actually get GCed after starting a new life and get reported, where in a real vat, the first incarnation would be gone, and not report anything.

I'm wondering if we can somehow silence this case in tests here (and in watch?). Maybe an integration of fake liveslots with the unhandled rejection tracker. No need to address now, but please add a comment and file an issue for follow-up.